### ISSUE TITLE
h2_fakesec_test_disappearing_server fails consistently on Linux

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#8388]
https://grpc-testing.appspot.com/job/gRPC_master_linux/160/testReport/junit/(root)/c_linux_opt/bins_opt_h2_fakesec_test_disappearing_server_GRPC_POLL_STRATEGY_epoll/

```
D1016 11:26:20.665918234    6560 test_config.c:277]          test slowdown: machine=1.000000 build=1.000000 total=1.000000
I1016 11:26:20.666122031    6560 ev_epoll_linux.c:86]        epoll engine will be using signal: 40
D1016 11:26:20.666143262    6560 ev_posix.c:106]             Using polling engine: epoll
I1016 11:26:20.678555347    6560 disappearing_server.c:194]  disappearing_server_test/chttp2/fake_secure_fullstack
I1016 11:26:20.680459283    6560 server.c:452]               Disconnected client: {"created":"@1476617180.680437316","description":"Last stream closed after sending GOAWAY","file":"src/core/ext/transport/chttp2/transport/chttp2_transport.c","file_line":1479}
E1016 11:26:20.681735663    6560 cq_verifier.c:241]          cq returned unexpected event: OP_COMPLETE: tag:0x1 OK
E1016 11:26:20.681757047    6560 cq_verifier.c:248]          expected tags:
0x65 GRPC_OP_COMPLETE result=1 test/core/end2end/tests/disappearing_server.c:140
```

**Comment by MrMoose** [comment]
Yes, this looks very familiar. I am having those crashes on my Linux Text Box here and initially thought it was something like #7608 but this call stack looks like what I was seeing.

Is there a workaround to this? Looks to me as if it is polling strategy dependent. If so, can I switch to another?

**Comment by MrMoose** [comment]
Here's a stacktrace that I just got.


[New LWP 3579]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `bin/Debug/mydd'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f8fad850428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
54      ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
[Current thread is 1 (Thread 0x7f8fa586a700 (LWP 3797))]
(gdb) bt
#0  0x00007f8fad850428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007f8fad85202a in __GI_abort () at abort.c:89
#2  0x00007f8fad8927ea in __libc_message (do_abort=do_abort@entry=2,
    fmt=fmt@entry=0x7f8fad9ab2e0 "*** Error in `%s': %s: 0x%s ***\n")
    at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007f8fad89ae0a in malloc_printerr (ar_ptr=<optimized out>, ptr=<optimized out>,
    str=0x7f8fad9ab3f0 "double free or corruption (out)", action=3) at malloc.c:5004
#4  _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:3865
#5  0x00007f8fad89e98c in __GI___libc_free (mem=<optimized out>) at malloc.c:2966
#6  0x000000000073c8b1 in grpc::Server::SyncRequest::CallData::~CallData() ()
#7  0x000000000073c1b9 in grpc::Server::RunRpc() ()
#8  0x0000000000741d87 in grpc::DynamicThreadPool::ThreadFunc() ()
#9  0x0000000000741a70 in grpc::DynamicThreadPool::DynamicThread::ThreadFunc() ()
#10 0x00007f8fae1bbc80 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#11 0x00007f8faed4170a in start_thread (arg=0x7f8fa586a700) at pthread_create.c:333
#12 0x00007f8fad92182d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
(gdb)

**Comment by jboeuf** [comment]
@jtattermusch How long has this test been failing?  We haven't touched the fake transport security for more than one year at least (unless they've been some changes I am not aware of).

**Comment by jtattermusch** [comment]
@jboeuf  Hard to tell, but for quite a while I'd say. 
Are you able to reproduce this locally?

**Comment by ctiller** [comment]
I have a PR out to mark it flaky right now. It's buried in with some other
stuff that should land today. This test really needs retry to be
implemented.

**Comment by MrMoose** [comment]
In case my stack trace really is the same problem:
I cannot tell for sure how long this occurred. Our server is only entering productive state. We started out with grpc 0.13 and I cannot recall anything like it. The first occurrences I remember started when I went to the 1.0.x branch before 1.0 was tagged. I upgraded a few times ever since and it didn't help.

On a side note, I've only seen it happening on Linux/clang. Never on Windows.
